scripts/player.py

```diff
@@ -1,24 +1,65 @@
 from abc import abstractmethod
+from math import exp, log
+from random import random
 
 from scripts.Qlearning import Qlearning
 
-class Player:
-    def __init__(self, **params ):
+
+class Player(object):
+    def __init__(self, **params):
         # params: type --> Dict[str, float]
         # params: alpha, T and so on
         self.Q_learning = Qlearning()
         self.params = params
 
-    @abstractmethod
-    def decide(self, game_skeleton):
-        pass
+        @abstractmethod
+        def decide(self, game_skeleton):
+            pass
+
 
 class VirtualPlayer(Player):
     pass
 
+
 class ModelAsPlayer(Player):
     '''uses Q-learning module, holds responsibility for updating max-likelihood function, and optimizing its parameters'''
-#     TODO: super init
+
+    def __init__(self):
+        super(ModelAsPlayer, self).__init__()
+        self.T = 0.5
+        self.decisions = []
+        self.Q_table = self.Q_learning.Q_table
+        # self.Q_table = [0.1, 0.4, 0.3, 0.2, 0.1, 0.4] # przykładowe
+
+    def probability_A(self, Q_A, Q_B):
+        p_A = 1 / (1 + exp((Q_A - Q_B) / self.T))
+        return p_A
+
     def decide(self, game_skeleton):
-        return 90 * [1]
+        condition_left = game_skeleton['StimulusLeft']
+        condition_right = game_skeleton['StimulusRight']
+        for index in range(len(condition_left)):
+            p_a = self.probability_A(self.Q_table[condition_left[index] - 1],
+                                     self.Q_table[condition_right[index] - 1])
+            dec = lambda x: random() < x
+            self.decisions.append(int(dec(p_a)))
+        return self.decisions
+
+    def max_likelihood_method(self, game_skeleton):
```

Comment:
I think it would be better to extract value of max likelihood function to a class' field and to rewrite this method without for as the same iteration happens in decide, here you could only leave lines 53,55,56 and invoke this method inside 'decide'

Reply:
and then change name to 'update_max_likelihood' or sth similar